app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,13 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!
   def index
```

Comment:
- [ ] Please use CanCanCan for Post authorization in Post Controller, you can do this by adding this line to postController `load_and_authorize_resource`